### ISSUE TITLE
Doxygen parsing does not recognize upper case HTML tags

### ISSUE THREAD

**Issue by tcottin** [llvm/llvm-project#83981]
Usually HTML tags are treated case insensitive.
But the Clang doxygen parsing does not treat HTML tags case insensitive.

This can be seen when activating -Wdocumentation and providing malformed HTML.

For example:

```C
// func.h

/// <b>
void func(void);
```

Running `clang -fsyntax-only -Wdocumentation func.h` gives the warning:

```
warning: HTML tag 'b' requires an end tag [-Wdocumentation-html]
    1 | /// <b>
      |     ~^~
1 warning generated.
```

Changing the tag to upper case:

```C
// func.h

/// <B>
void func(void);
```

does not give the warning.

I am using:

```
> clang --version
Ubuntu clang version 17.0.6 (++20231209124227+6009708b4367-1~exp1~20231209124336.77)
Target: x86_64-pc-linux-gnu
Thread model: posix
InstalledDir: /usr/bin
```

For my use case I fixed this by adapting the TableGen generated functions from `clang/utils/TableGen/ClangCommentHTMLTagsEmitter.cpp` to compare the passed tag names using lower casing.

I could push this upstream if the community is interested.
